### ISSUE TITLE
Tests should run in parallel

### ISSUE THREAD

**Issue by inducer** [illinois-ceesd/mirgecom#28]
I've had good experiences with `pytest-xdist` with `-n 4`. Github's CI machines give us two cores... we should make use of them.

**Comment by matthiasdiener** [comment]
I am not sure the CI machines can handle the memory requirements though...

**Comment by inducer** [comment]
I've ([mostly](https://gitlab.tiker.net/inducer/pytential/-/issues/146)) not had any issues with memory consumption.

**Comment by matthiasdiener** [comment]
Ok, but I have seen some weird CI failures that might be explained by memory issues:
https://github.com/illinois-ceesd/mirgecom/pull/31/checks?check_run_id=916655177

```
============================= test session starts ==============================
platform linux -- Python 3.8.5, pytest-5.4.3, py-1.9.0, pluggy-0.13.1
rootdir: /home/runner/work/mirgecom/mirgecom
/home/runner/work/_temp/7628ae5a-ea45-46c1-9650-3e604de5ab46.sh: line 4:  6679 Aborted                 (core dumped) python -m pytest --durations=10 --tb=native --junitxml=pytest.xml --doctest-modules -rxsw .
```

**Comment by inducer** [comment]
Argh, that's annoying. Could you look into making sure we get a Python-side and C-side backtrace when this happens?

**Comment by inducer** [comment]
Also, pytest captures output by default. It can be told not to with `--capture=no`.

**Comment by matthiasdiener** [comment]
> `--capture=no`.

Seems to make no difference.

> Argh, that's annoying. Could you look into making sure we get a Python-side and C-side backtrace when this happens?

If I understood https://github.com/pytest-dev/pytest-faulthandler and https://blog.richard.do/2018/03/18/how-to-debug-segmentation-fault-in-python/ correctly, the backtrace should happen automatically. Not sure what's going wrong.

**Comment by matthiasdiener** [comment]
Rerunning the CI in that PR has slightly more helpful output:

```
============================= test session starts ==============================
platform linux -- Python 3.8.5, pytest-5.4.3, py-1.9.0, pluggy-0.13.1
rootdir: /home/runner/work/mirgecom/mirgecom
--------------------------------------------------------------------------
[[10542,1],0]: A high-performance Open MPI point-to-point messaging module
was unable to find any relevant network interfaces:

Module: OpenFabrics (openib)
  Host: fv-az68

Another transport will be used instead, although this may result in
lower performance.

NOTE: You can disable this warning by setting the MCA parameter
btl_base_warn_component_unused to 0.
--------------------------------------------------------------------------
Not enough memory to run on this device.
/home/runner/work/_temp/ff589edc-ba3e-45ba-982b-a843b4c6ef1e.sh: line 4:  6347 Aborted                 (core dumped) python -m pytest --capture=no --durations=10 --tb=native --junitxml=pytest.xml --doctest-modules -rxsw .
```
 
> `Not enough memory to run on this device.`

Appears to come from pocl (https://github.com/pocl/pocl/blob/master/lib/CL/devices/common.c#L1182)

**Comment by inducer** [comment]
> Not enough memory to run on this device.

I know that error message (unfortunately!). What it's trying (and failing) to say is that there are two different versions of libhwloc being linked in. (pocl depends on it, as does ompi) So this should be fixable by twiddling conda dependencies.

**Comment by matthiasdiener** [comment]
It seems that explicitly installing hwloc v1 instead of v2 fixes this issue. Is this an acceptable workaround?

**Comment by inducer** [comment]
Yes, totally. I do it, too: https://www.news-gazette.com/the-news-gazette-has-questionable-news/article_1f813a3f-efdb-5b62-b6a3-0147c43b35e4.html

**Comment by inducer** [comment]
@MTCam How come this leaves you unhappy? It's clear that the libhwloc people screwed up by breaking ABI without appropriately changing the soname. It sucks, but IMO this is the best we can do.

**Comment by MTCam** [comment]
> @MTCam How come this leaves you unhappy? It's clear that the libhwloc people screwed up by breaking ABI without appropriately changing the soname. It sucks, but IMO this is the best we can do.

Not unhappy as much as terrified.   Three thumbs down for the News-Gazette story linked in your post.

**Comment by inducer** [comment]
Aaargh. That was not what I was meaning to link. Copy-paste fail. Fixed my link, sorry about the off-topic noise.
